### PR TITLE
Sphinx setup for sdk docs generation

### LINKED ISSUE #2373: [GSOC] Poject 7 Tracking Issue: Automating Documentation for katib Python SDK
### Objective
Automate the Katib Python SDK reference documentation generation and render it on `kubeflow.org`.

### Tasks
- [x] **Proposal Submission:** Submit the GSOC project proposal as a project proposal in Katib.
- [x] **Review Existing Documentation:** Analyze documentation from other projects for best practices.
- [x] **Local Katib Setup:** Clone Katib repository for kickoff project development.
- [x] **Evaluate Documentation Tools:** Compare Sphinx, MkDocs, Jupyter Book, etc., to choose the best tool for Katib's needs.
- [x] **Configure and Test Documentation Tool:** Set up Sphinx (or chosen tool) for Katib. Could you make sure of the quality and readability of the generated documentation? ([#2392](https://github.com/kubeflow/katib/pull/2392))
- [ ] **Integrate with Kubeflow's Website Repo:** Make the Katib repository a submodule of the website repository and test documentation sharing.
- [ ] **Automate Documentation Updates:** 
  - Write a GitHub Action to automate documentation updates and trigger documentation sharing to the website repository.
  - Script to handle events and update website files.

### Expected Outcomes
- Fully automated, high-quality documentation process.
- Up-to-date documentation on the Katib website.

### PR DESCRIPTION
**Pull request by shivas1516**
**What this PR does / why we need it**:
This PR sets the Sphinx configuration for generating the Katib SDK API reference documentation. It also includes updates to `gen_python.py` to build the documentation in Markdown format.

**Which issue(s) this PR fixes** _(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)_:
Fixes  #2373

Demo Link: https://katib-demo.readthedocs.io/en/latest/index.html

Note: Before merging this into the main branch, we must create a new Read the Docs project and import the katib repo to it.

Error:
Screenshot from 2024-08-30 22-59-26

seeking for suggestions to resolve this error

**Checklist:**

- [ ] [Docs](https://www.kubeflow.org/docs/components/katib/) included if any changes are user facing

### CODE REVIEW

File: hack/gen-python-sdk/post_gen.py [hunk 1]
```diff
@@ -183,6 +184,29 @@ def update_python_sdk(src, dest, versions=("v1beta1")):
...
+        print(f"Error building Sphinx documentation: {e}")
+    except FileNotFoundError:
+        print("Error: 'make' command not found. Please ensure you have Sphinx installed.")
+
 
 if __name__ == "__main__":
     update_python_sdk(src=sys.argv[1], dest=sys.argv[2])
+    
+    # SDK docs update
+    upgate_python_sdk_docs()
```
Comment (shivaylamba) [inline hack/gen-python-sdk/post_gen.py:212]: update

File: hack/gen-python-sdk/post_gen.py [hunk 2]
```diff
@@ -183,6 +184,29 @@ def update_python_sdk(src, dest, versions=("v1beta1")):
     # Clear working dictionary
     shutil.rmtree(src)
 
+def upgate_python_sdk_docs():
```
Comment (shivaylamba) [inline hack/gen-python-sdk/post_gen.py:187]: update

File: sdk/python/v1beta1/README.md [hunk 3]
```diff
@@ -1,125 +1,53 @@
...
-- [V1beta1TrialTemplate](docs/V1beta1TrialTemplate.md)
-
-## Documentation For Authorization
-
-All endpoints do not require authorization.
-
-## Author
-
-prem0912
+Please follow the [examples](https://github.com/shivas1516/katib/tree/master/sdk) to learn more about Katib SDK.
```
Comment (shivaylamba) [inline sdk/python/v1beta1/README.md:49]: THis has to be replaced with the actual github repo

File: sdk/python/v1beta1/README.md [hunk 4]
```diff
@@ -1,125 +1,53 @@
...
-Our SDK is located in [`kubeflow-katib` PyPi package](https://pypi.org/project/kubeflow-katib/).
-Katib Python SDK is published as part of the Katib patch releases.
-You can check the release process [here](../../../scripts/v1beta1/release.sh).
-For each Katib patch release, we upload a new SDK version to the PyPi.
-The SDK version is equal to the Katib version.
+The Katib Python SDK is released as part of Katib's patch releases. For each patch release, a new version of the SDK is uploaded to PyPi. The SDK version corresponds directly to the Katib version.
+
+You can find the Katib Python SDK in the [`kubeflow-katib` package on PyPi](https://pypi.org/project/kubeflow-katib/).
+
+For detailed information on the release process, please refer to the [release script](https://github.com/shivas1516/katib/blob/master/scripts/v1beta1/release.sh) in the Katib repository.
```
Comment (shivaylamba) [inline sdk/python/v1beta1/README.md:44]: needs to be replaced with main repo

**Review by shivaylamba — CHANGES_REQUESTED** [review]

**Comment by shivaylamba** [comment]
Also ensure if you can pass the failing linting and go tests

File: hack/gen-python-sdk/post_gen.py (hunk 2 above)
Comment (shivas1516) [reply hack/gen-python-sdk/post_gen.py]: @shivaylamba I’ve removed the `update_python_sdk_docs()` method. It appears that Read the Docs updates the documentation automatically using the `.readthedocs.yaml` configuration, so there's no need for a separate workflow.

File: sdk/python/v1beta1/README.md (hunk 3 above)
Comment (shivas1516) [reply sdk/python/v1beta1/README.md]: updated in the recent push

File: sdk/python/v1beta1/README.md (hunk 4 above)
Comment (shivas1516) [reply sdk/python/v1beta1/README.md]: updated in the recent push, check and let me know if addition changes sir

**Comment by shivas1516** [comment]
> @shivas1516 Please can you rise a PR for Training Operator as well since it doesn't have this problem with gRPC import package ?

Sure, I configured the Sphinx and completed the docs for the client but most of the docstrings in each models are not  friendly to the Sphinx and need to be manually altered it.

demo link: https://testtrainingoperator.readthedocs.io/en/latest/index.html

**Pull request closed without merging**